sdk/src/azure/core/az_mqtt5_rpc_client_hfsm.c

```diff
@@ -664,8 +666,11 @@ static az_result faulted(az_event_policy* me, az_event event)
   {
     case AZ_HFSM_EVENT_ENTRY:
     {
-      _az_RETURN_IF_FAILED(az_event_policy_send_inbound_event(
-          me, (az_event){ .type = AZ_HFSM_EVENT_ERROR, .data = NULL }));
+      if (az_result_failed(az_event_policy_send_inbound_event(
+              me, (az_event){ .type = AZ_HFSM_EVENT_ERROR, .data = NULL })))
+      {
+        az_platform_critical_error();
```

Comment:
Not sure if we need this on the faulted entry - the root one is more of a catch all if errors are bubbling up to that level (which they generally shouldn't because they should be handled in a sub-state), the application needs to be notified, potentially forcefully. I think a single hfsm going to faulted doesn't necessarily require the same force of notification

Reply:
I misinterpreted the earlier comment, reverted the changes for faulted in this and the other HFSM files